src/librustc_mir/transform/check_consts/mod.rs

```diff
@@ -0,0 +1,46 @@
+use rustc::hir::def_id::DefId;
+use rustc::mir;
+use rustc::ty::{self, TyCtxt};
+
+pub use self::qualifs::Qualif;
+
+pub mod ops;
+mod qualifs;
+mod resolver;
+pub mod validation;
+
+/// Information about the item currently being validated, as well as a reference to the global
+/// context.
+pub struct Item<'mir, 'tcx> {
```

Comment:
The name of the equivalent data structure in the old code was `ConstCx`, although it was also used for doing promotion in  `NonConstFn`s. I like `Item` better, but I agree it's still pretty vague.

Reply:
Maybe `ItemValidationInfo`?

Reply:
This type appears in a lot of function signatures (even more once [this alias](https://github.com/rust-lang/rust/pull/64470/files#diff-5f5eb5bcac745676f312365558e20ba8R7) disappears), so brevity is desirable.

Reply:
A bit shorter `CheckCtx`?

Reply:
I defer to others, but none of the alternatives seem better to me. Writing `item.body` and `item.tcx` are clearer to me than `cx.body` and `cx.tcx`.

Reply:
`icx: ItemCtxt`? Just like we have `fcx: FnCtxt`

Reply:
In that spirit, maybe `ccx: ConstCtxt` cause it signifies the analysis rather than the item.

Reply:
but `fcx` signifies a function context, so `icx` being an item context seems fine :laughing:

I don't have an opinion either way, just wanted to say why it would be consistent.

Reply:
> just wanted to say why it would be consistent.

👍 -- you are right. :)